builtin/credential/jwt/cli.go

```diff
@@ -69,19 +72,45 @@ func (h *CLIHandler) Auth(c *api.Client, m map[string]string, nonInteractive boo
 		port = defaultPort
 	}
 
+	var serverURL *url.URL
+	callbackMode, ok := m[FieldCallbackMode]
+	if !ok {
+		callbackMode = defaultCallbackMode
+	} else if callbackMode == "direct" {
+		serverAddr := os.Getenv("BAO_ADDR")
+		if serverAddr == "" {
+			serverAddr = os.Getenv("VAULT_ADDR")
+		}
+		if serverAddr != "" {
+			serverURL, _ = url.Parse(serverAddr)
+		}
+	}
+
 	callbackHost, ok := m[FieldCallbackHost]
 	if !ok {
-		callbackHost = defaultCallbackHost
+		if serverURL != nil {
+			callbackHost = serverURL.Hostname()
+		} else {
+			callbackHost = defaultCallbackHost
+		}
 	}
 
 	callbackMethod, ok := m[FieldCallbackMethod]
 	if !ok {
-		callbackMethod = defaultCallbackMethod
+		if serverURL != nil {
+			callbackMethod = serverURL.Scheme
+		} else {
+			callbackMethod = defaultCallbackMethod
+		}
 	}
 
 	callbackPort, ok := m[FieldCallbackPort]
 	if !ok {
-		callbackPort = port
+		if serverURL != nil {
+			callbackPort = serverURL.Port() + "/v1/auth/" + mount
```

Comment:
I don't know anything about namespaces, and it doesn't otherwise occur in cli.go.  There is another case in cli.go that already uses "/v1/auth" with mount so I think we should stick with the precedent.

Reply:
Ah, I guess namespaces could just be specified as part of the mount itself, I think. So I think we're probably good here. Thanks @DrDaveD!